src/hooks/chats.ts

```diff
@@ -0,0 +1,193 @@
+import { useMemo } from "react";
+
+import { CHAT_TYPES, SetAnyChatTabOptions } from "types/Chat";
+
+import {
+  userPrivateChatsSelector,
+  chatUsersByIdSelector,
+  chatUIStateSelector,
+} from "utils/selectors";
+import { chatSort } from "utils/chat";
+import { filterUniqueKeys } from "utils/filterUniqueKeys";
+import { hasElements } from "utils/types";
+
+import { setChatTab, setChatSidebarVisibility } from "store/actions/Chat";
+
+import { isLoaded, useFirestoreConnect } from "./useFirestoreConnect";
+import { useSelector } from "./useSelector";
+import { useUser } from "./useUser";
+import { useDispatch } from "./useDispatch";
+
+import { DOCUMENT_ID, NUM_CHAT_UIDS_TO_LOAD } from "settings";
+
+/////////////// SHARED CHATS HOOKS ///////////////
+
+export const useChatControls = () => {
+  const dispatch = useDispatch();
+
+  const openChat = (
+    chatOptions: SetAnyChatTabOptions = { chatType: CHAT_TYPES.VENUE_CHAT }
+  ) => {
+    dispatch(setChatSidebarVisibility(true));
+    dispatch(setChatTab(chatOptions));
+  };
+
+  const closeChat = () => {
+    dispatch(setChatSidebarVisibility(false));
+  };
+
+  const { isChatSidebarVisible, openedChatType } = useSelector(
+    chatUIStateSelector
+  );
```

Comment:
Once more things are added to this redux store, we are going to want to use a far more 'refined' selector. As this is written currently. anytime anything in the chat state changes, this hook has to re-run, only to then ignore everything except for `isChatSidebarVisible`, `openedChatType`.

We want our selector function to return the absolute minimal amount of data possible to satisfy our needs. Selectors can also be composed, so you don't need to write each one from the 'root store'.